conda/gateways/repodata/__init__.py

```diff
@@ -513,7 +512,7 @@ def __setitem__(self, key: str, item: Any) -> None:
         if key in self._aliased:
             key = key[1:]  # strip underscore
         if key in self._strings and not isinstance(item, str):
-            warnings.warn('Replaced non-str RepodataState[{key}] with ""')
+            warnings.warn(f'Replaced non-str RepodataState[{key}] with ""')
```

Comment:
Two things:

- Should we also log the `item` that got replaced?
- Should this be `logging.warning` instead?

Reply:
I consulted with @sven6002 and @kenodegard and we think it makes more sense for now to only use the `warnings` module exclusively for deprecations and `logging` everywhere else. The code has been updated accordingly!